### DISCUSSION TITLE
map alignment

### DISCUSSION THREAD

**Discussion by pac0388** [OpenMS/OpenMS#5271] [Q&A]
Hi, can anyone point me in the direction of example code for map alignment for pyopenms? Based on the documentation I can figure out how to make feature maps and then how to work with a consensus map, but I don't see how to get from a group of feature maps to the consensus map using something like MapAlignmentAlgorithmPoseClustering. Ideally I could loop over a group of feature maps in a directory, load them into a consensus map, and then apply the clustering algorithm. Any help would be greatly appreciated.

**Comment by jpfeuffer** [comment]
Hi!
Good question. It could be something like:

```python
## your reference map in ref_fm (e.g. pick the one with most features)
## your remaining list(FeatureMaps) in fms

ma = MapAlignmentAlgorithmPoseClustering()
# optional: set any parameters on ma
ma.setReference(ref_fm)
for fm in fms:
  trafo = TransformationDescription() # to be filled
  ma.align(fm, trafo)
  MapAlignmentTransformer().transformRetentionTimes(fm, trafo);
```

Disclaimer: untested.

**Reply by jpfeuffer** [reply]
I took it a bit from here:
https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/pyTOPP/MapAlignerPoseClustering.py

**Reply by pac0388** [reply]
This is exactly what i needed, thank you

**Comment by timosachsenberg** [comment]
Another possibility is to install OpenMS and call the command line tool from python in a process.
